flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -405,36 +374,13 @@ class JobManager(
 
       currentResourceManager match {
         case Some(rm) =>
-          val future = (rm ? decorateMessage(new RegisterResource(taskManager, msg)))(timeout)
-          future.onComplete {
-            case scala.util.Success(response) =>
-              // the resource manager is available and answered
-              self ! response
-            case scala.util.Failure(t) =>
-              t match {
-                case _: TimeoutException =>
-                  log.info("Attempt to register resource at ResourceManager timed out. Retrying")
-                case _ =>
-                  log.warn("Failure while asking ResourceManager for RegisterResource. Retrying", t)
-              }
-              // slow or unreachable resource manager, register anyway and let the rm reconnect
-              self ! decorateMessage(new RegisterResourceSuccessful(taskManager, msg))
-              self ! decorateMessage(new ReconnectResourceManager(rm))
-          }(context.dispatcher)
-
+          log.info(s"Register task manager $resourceId at the resource manager.")
+          rm ! decorateMessage(new RegisterResource(msg))
```

Comment:
I just don't understand why you remove this functionality. It was not broken in any way. Of course, we can always add features later (that is true for any component) but it changes the original RM design. If we want to add monitoring of the pool size later on, we will have to re-add the proper registration at the RM.